src/main/java/com/practice/slideshow/service/KafkaLoggingService.java

```diff
@@ -19,34 +21,40 @@
 @Slf4j
 public class KafkaLoggingService {
 
-  private final KafkaTemplate<String, String> kafkaTemplate;
+  private final KafkaTemplate<String, ImageResponse> kafkaTemplate;
```

Comment:
it should be string json or string string. You are not using avro as result there could be some issues